kitsune/wiki/views.py

```diff
@@ -435,7 +435,89 @@ def steal_lock(request, document_slug, revision_id=None):
 @require_http_methods(["GET", "POST"])
 @login_required
 def edit_document(request, document_slug, revision_id=None):
-    """Create a new revision of a wiki document, or edit document metadata."""
+    """Create a new revision of a wiki document"""
+    user = request.user
+
+    doc = get_visible_document_or_404(
+        user,
+        locale=request.LANGUAGE_CODE,
+        slug=document_slug,
+        look_for_translation_via_parent=True,
+        return_parent_if_no_translation=True,
+    )
+
+    if doc.locale != request.LANGUAGE_CODE:
+        # We've fallen back to the parent, since no visible translation existed.
+        url = reverse("wiki.translate", locale=request.LANGUAGE_CODE, args=[document_slug])
+        return HttpResponseRedirect(url)
+
+    # If this document has a parent, then the edit is handled by the
+    # translate view. Pass it on.
+    if doc.parent:
+        return translate(request, doc.parent.slug, revision_id)
+    if revision_id:
+        rev = get_object_or_404(Revision, pk=revision_id, document=doc)
+    else:
+        rev = doc.current_revision or doc.revisions.order_by("-created", "-id")[0]
+
+    rev_form = None
+
+    if request.method == "GET":
+        if doc.allows(user, "create_revision"):
```

Comment:
You could reduce the level of nesting by using something like:

```python
if doc.allows(user, "create_revision"):
    rev_form = RevisionForm(request.POST or None, instance=rev, ...)

    # request.GET part goes here
    ....
    
    if rev_form.is_valid():  # request.POST goes here
        ....
```
of course this will require a restructure to the whole view but most probably it's going to improve readability and make it more compact.

If the above is not possible you could remove the check for the GET and just check for post

Reply:
hm. I'll look at this!  Thanks.